form.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"context"
+	"github.com/go-playground/locales/en"
+	ut "github.com/go-playground/universal-translator"
+	"github.com/go-playground/validator/v10"
+	en_translations "github.com/go-playground/validator/v10/translations/en"
+	"github.com/gorilla/schema"
+	"github.com/pdeguing/empire-and-foundation/data"
+	"github.com/pdeguing/empire-and-foundation/ent/user"
+	"reflect"
+	"strings"
+)
+
+var decoder *schema.Decoder
+var validate *validator.Validate
+var translator ut.Translator
+
+func init() {
```

Comment:
The validation library uses reflection to see what fields are in a struct. It will cache that in the decoder. So I don't want to recreate all that stuff on every request. `RegisterTagNameFunc` is used to get the name of the field from the `name` tag in the struct.

+ some translations so we get sensible error messages.
+ custom rule to check if the username is not already used for registration.
+ translations for the custom rule.